docs/02-app/01-building-your-application/04-caching/index.mdx

```diff
@@ -0,0 +1,468 @@
+revalidateTag('a')
+```
+
+There are two places you can use `revalidateTag`, depending on what you're trying to achieve:
+
+1. [Route Handlers](/docs/app/building-your-application/routing/router-handlers) - to revalidate data in response of a third party event (e.g. webhook).
+2. [Server Actions](/docs/app/building-your-application/data-fetching/server-actions) - to revalidate data after a user action (e.g. form submission).
+
+{/_ TO DO: Talk about how revalidateTag affects other caches and re-rendering.
+If you're calling it from a route handler, then it makes sense if it's for next UI visit, but for server actions, the user would expect a visual update / rerender _/}
+
+### `revalidatePath`
+
+`revalidatePath` allows you manually revalidate data **and** re-render the route segments below a specific path in a single operation. Calling `revalidatePath` methods revalidate the Data Cache, which in turn invalidates the Full Route Cache. It does not immediately invalidate the Router Cache due to the [automatic invalidation period](#duration-and-invalidation).
+
+```jsx
+revalidatePath('/')
+```
+
+There are two places you can use `revalidatePath`, depending on what you're trying to achieve:
+
+1. [Route Handlers](/docs/app/building-your-application/routing/router-handlers) - to revalidate data in response of a third party event (e.g. webhook).
+2. [Server Actions](/docs/app/building-your-application/data-fetching/server-actions) - to revalidate data after a user action (e.g. form submission).
+
+> ** `revalidatePath` vs. `router.refresh`**:
+>
+> Calling `router.refresh` will clear the Router cache, and re-render route segments on the server without invalidating the Data Cache or the Full Route Cache.
+>
+> The difference is that `revalidatePath` **revalidates** cached data in Static Routes, whereas `router.refresh` **reuses** cached data.
+
+See the [`revalidatePath` API reference]() for more information.
+
+> **Differences between `res.revalidate` vs. `revalidateTag` / `revalidatePath`**:
+>
+> There are a couple of differences between `res.revalidate` and `revalidateTag` / `revalidatePath`:
+>
+> 1. `res.revalidate` was built for API Routes in the Pages Router, when rendering and data fetching were coupled together in the same cache.
+> 2. `res.revalidate` applies specifically to the Full Route Cache, and not the Data Cache or Router Cache.
+> 3. `res.revalidate` returns a promise, whereas `revalidateTag` and `revalidatePath` are non-blocking, _fire-and-forget_ APIs.
+
+### Dynamic Functions
+
+`cookies`, `headers`, `useSearchParams`, and `searchParams` are all dynamic functions that depend on runtime information. Using them will opt a route out of the Full Route Cache, in other words, the routes will be dynamically rendered.
+
+#### `cookies`
+
+Using `cookies.set` or `cookies.delete` in a Server Action invalidates the Router Cache to prevent routes that use cookies from becoming stale (e.g. authentication).
```

Comment:
```suggestion
Using `cookies.set` or `cookies.delete` in a Server Action invalidates the Router Cache to prevent routes that use cookies from becoming stale (e.g. to reflect authentication changes).
```